### PR TITLE
Add google-maps support to ios

### PR DESCRIPTION
**Pull request by wilsonpage**
- Required to be able to style maps.
- Cocoa pods support added to ease mega pain of importing third-party
  libraries. It's a 100x simpler to install pods than manual methods,
but I don't know if there's a catch?
- The use of CocoaPods requires that you open `MagnetJourney.xcworkspace` instead of `MagnetJourney.xcodeproj` in XCode. So close the window and reopen otherwise builds will fail.

### CODE REVIEW

**Comment by wilsonpage** [comment]
@gmarty @arcturus r?

File: ios/Podfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+source 'https://github.com/CocoaPods/Specs.git'
+platform :ios, '9.0'
+
+target 'MagnetJourney' do
+  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
+  # use_frameworks!
+  
+  pod 'React', path: '../node_modules/react-native', :subspecs => [
```
Comment (arcturus) [inline ios/Podfile:8]: Why do we need to specify the React native classes as a pod?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - React/Core
+  - React/RCTVibration (0.40.0):
+    - React/Core
+  - React/RCTWebSocket (0.40.0):
+    - React/Core
+  - React/yoga (0.40.0)
+
+DEPENDENCIES:
+  - GoogleMaps
+  - React/Core (from `../node_modules/react-native`)
```
Comment (arcturus) [inline ios/Podfile.lock:38]: Oh I see, the dependencies in GoogleMaps pod .... did I say that I hate pods :(

File: js/components/MagnetMapView/index.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import React, { Component, PropTypes } from 'react';
```
Comment (arcturus) [inline js/components/MagnetMapView/index.js:1]: +1 to move each component to it's own directory.

**Comment by arcturus** [comment]
Just reviewed the podfile, the AppDelegate and the changes in the js component.

All of them looking good for me. Personally I don't like cocoapods, it mess around with your project layout and the way of having to commit everything to the main repo seems obsolete (specially if you need to update pods etc.) I do prefer Carthage ... but with that said, if this is the way of having gmaps support using the airmaps, let's make it happen.

Please rebase and shiiiiiip! @gmarty do you agree?

File: ios/Podfile (hunk 1 above)
Comment (wilsonpage) [reply ios/Podfile:8]: The React libraries are no longer added to the 'Build Phases' tab and defined here instead. I copied the airbnb-maps example app.

File: ios/Podfile.lock [hunk 4]
```diff
@@ -0,0 +1,60 @@
+PODS:
```
Comment (wilsonpage) [inline ios/Podfile.lock:1]: This file is auto generated

**Comment by wilsonpage** [comment]
I'm currently checking in the CocoaPod dependencies, that's what is recommended on the official site. Although it does add a lot of noise to this diff!

**Comment by wilsonpage** [comment]
Damn Github UI, always close PRs by mistake!

**Comment by gmarty** [comment]
Do we need to keep all the files from the `ios/Pods/GoogleMaps/Example/` folder?
It seems that they are mostly samples and increase the size of our repo.

**Comment by wilsonpage** [comment]
@gmarty we could gitignore, but it could end up being an endless game of cat and mouse as these are internal details of third-party packages.

**Pull request merged**